### ISSUE TITLE
chezmoiexternal with subdir config inconsistent state

### ISSUE THREAD

**Issue by QCU266** [twpayne/chezmoi#3131]
## What exactly are you trying to do?

I used [neovim nvchad](https://nvchad.com/docs/quickstart/install), however nvchad [custom config script should in subdir `lua/custom`](https://nvchad.com/docs/config/walkthrough#custom_config). 

so, I want to use `.chezmoiexternal.toml` to manager nvchad offical repo code , and to use `dot_config/nvim/lua/custom` to manager myself config. 

However, I got `inconsistent state` error:

```bash
$ chezmoi apply
chezmoi: .config/nvim: inconsistent state (/Users/qcu/.local/share/chezmoi/dot_config/nvim, https://github.com/NvChad/NvChad defined in /Users/qcu/.local/share/chezmoi/.chezmoiexternal.toml)
```
How to configure this scenario？

## What have you tried so far?

Describe what you have tried so far.

## Where else have you checked for solutions?

* [x] I have read [chezmoi's user guide](https://chezmoi.io/user-guide/command-overview/), and not found the answer.
* [x] I have searched [chezmoi's reference guide](https://chezmoi.io/reference/), and not found the answer.

**Comment by halostatue** [comment]
#3065 and #2015 appear to be related to what you’re reporting. You may want to skim those to see if any of the solutions there suggest themselves for this issue.

If they do not, please include your `.chezmoiexternal` configuration and the output of `chezmoi doctor` as well as any other information you discover while looking over those.

**Comment by halostatue** [comment]
I’m glad you found at least a partial solution here, but as I am not using externals in this way, I am the wrong person to answer those questions. Hopefully someone else will be able to chime in on this.

**Comment by twpayne** [comment]
tl;dr if you want to add your own files to an external directory, use `type = "archive"`.

> However, I also encountered some confusing aspects. I'm not sure if it's a bug or intentionally designed this way.

I agree that this is confusing. There are good design reasons behind the decision. What's happening is:

When the external `type = "archive"`, then chezmoi downloads the archive and gets a full, precise view of the state of every file. This means that chezmoi can merge this state with its own state (e.g. the extra `dot_config/nvim/lua/custom` file) and get a consistent view of what you want in your home directory.

When the external `type = "git-repo"`, then chezmoi delegates management of that directory to git. git is a large, complex application with many subtleties, and chezmoi cannot hope to follow git's behavior exactly. So, chezmoi does not attempt to merge its state, which is why you get the "inconsistent state" error.

So, your use of `type = "archive"` is the right approach here.
